nomos-http-api/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use overwatch_rs::{
+    services::{
+        handle::ServiceStateHandle,
+        relay::NoMessage,
+        state::{NoOperator, NoState},
+        ServiceCore, ServiceData,
+    },
+    DynError,
+};
+
+/// A simple abstraction so that we can easily
+/// change the underlying http server
+#[async_trait::async_trait]
+pub trait Backend {
+    type Error: std::error::Error + Send + Sync + 'static;
+    type Settings: Clone + Send + Sync + 'static;
+
+    async fn new(settings: Self::Settings) -> Result<Self, Self::Error>
+    where
+        Self: Sized;
+
+    async fn serve(self) -> Result<(), Self::Error>;
+}
+
+#[derive(Debug, Clone)]
+pub struct ApiServiceSettings<S> {
+    pub server_settings: S,
+}
+
+pub struct ApiService<E: Backend> {
+    settings: ApiServiceSettings<E::Settings>,
+}
+
+impl<E: Backend> ServiceData for ApiService<E> {
```

Comment:
```suggestion
pub struct ApiServiceSettings<S> {
    pub backend_settings: S,
}

pub struct ApiService<B: Backend> {
    settings: ApiServiceSettings<B::Settings>,
}

impl<B: Backend> ServiceData for ApiService<B> {
```
we could probably rename like this?

Reply:
And, `endpoint` -> `backend` at line 59.

Reply:
Nice catch!

Reply:
Thank you! I forgot to change the generic param.